x-pack/legacy/plugins/apm/public/components/app/Permission/index.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiButton,
+  EuiEmptyPrompt,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiPanel,
+  EuiTitle
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useMemo, useState } from 'react';
+import { matchPath } from 'react-router-dom';
+import styled from 'styled-components';
+import { FETCH_STATUS, useFetcher } from '../../../hooks/useFetcher';
+import { useLocation } from '../../../hooks/useLocation';
+import { fontSize, pct, px, units } from '../../../style/variables';
+import { ElasticDocsLink } from '../../shared/Links/ElasticDocsLink';
+import { SetupInstructionsLink } from '../../shared/Links/SetupInstructionsLink';
+import { routes } from '../Main/route_config';
+
+export const Permission: React.FC = ({ children }) => {
+  const [isPermissionPageEnabled, setIsPermissionsPageEnabled] = useState(true);
+
+  const { data, status } = useFetcher(callApmApi => {
+    return callApmApi({
+      pathname: '/api/apm/security/permissions'
+    });
+  }, []);
+
+  // Return null until receive the reponse of the api.
+  if (status === FETCH_STATUS.LOADING || status === FETCH_STATUS.PENDING) {
+    return null;
+  }
+  // When the user doenst have permission and he didnt escape, show the Permission page
+  if (!data?.hasPermission && isPermissionPageEnabled) {
+    return (
+      <PermissionPage
+        onEscapeHatchClick={() => setIsPermissionsPageEnabled(false)}
+      />
+    );
+  }
```

Comment:
Would `if (data?.hasPermission === false && isPermissionPageEnabled) {}` be enough? Explicitly checking for `false` means you don't need a status check as long as you don't set a default value.

Reply:
Do we need a 403 check here as well?

Reply:
I change it to `if (data?.hasPermission === false && isPermissionPageEnabled) {}` like you suggested, I'm checking if the status is 403 in the server, so it's not, hasPermission will be undefined and the APM page will be shown.

Reply:
I _think_ a 403 could happen before that, for instance if the user does not have the appropriate Kibana permissions. But for now this is probably good enough.